src/app/add-course-page/add-course-page/add-course-page.component.ts

```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
     this.authors = authors;
   }
 
-  updateCourse(): void {
-    this.course.creationDate = new Date(this.course.creationDate);
-    this.courseService.updateCourse(this.course);
+  async updateCourse(): Promise<void> {
```

Comment:
you don't need to do it async especially declare return type as Promise<void> when you don't return anything)
it should looks as something like this:
```
 updateCourse(): void {
    this.courseService.updateCourse(this.course).then(() => this.router.navigate(['/courses']));
  }
```